hugegraph-core/src/main/java/org/apache/hugegraph/util/Consumers.java

```diff
@@ -49,13 +49,13 @@ public final class Consumers<V> {
     private final ExecutorService executor;
     private final Consumer<V> consumer;
     private final Runnable done;
-
+    private final Consumer<Throwable> exceptionHandle;
     private final int workers;
+    private final List<Future> runnings;
     private final int queueSize;
     private final CountDownLatch latch;
-    private final BlockingQueue<V> queue;
-
-    private volatile boolean ending = false;
+    private final BlockingQueue<VWrapper<V>> queue;
+    private final VWrapper<V> queueEnd = new VWrapper(null);
```

Comment:
we can just define a mark like `private final static Object QUEUE_END = new Object();`